dspy/clients/lm.py

```diff
@@ -27,18 +24,41 @@
 
 GLOBAL_HISTORY = []
 
+
 class LM:
+    """
+    A language model supporting chat and completions requests for use with DSPy
+    modules and programs.
+    """
+
     def __init__(
-            self, 
-            model,
-            model_type='chat', 
-            temperature=0.0,
-            max_tokens=1000,
-            cache=True,
-            launch_kwargs=None,
-            callbacks=None,
-            **kwargs
-        ):
+        self,
+        model: str,
+        model_type: Literal["chat", "completions"] = "chat",
+        temperature: float = 0.0,
+        max_tokens: int = 1000,
+        cache: bool = True,
+        launch_kwargs: Optional[Dict[str, Any]] = None,
+        callbacks: Optional[List[BaseCallback]] = None,
+        num_retries: int = 8,
```

Comment:
Empirically, 8 retries with exponential backoff in LiteLLM takes just over 1 minute with LiteLLM's builtin / default retry strategy. I defer to their exponential backoff retry strategy defaults, rather than defining our own (presumably, they know best)